scripts/doc-checker.py

```diff
@@ -0,0 +1,240 @@
+"""
+Validate docstrings using pydocstyle and numpydoc.
+
+Example usage:
+python scripts/doc-checker.py asv_bench/benchmarks/utils.py modin/pandas
+"""
+
+import argparse
+import pathlib
+import subprocess
+import os
+import ast
+from typing import List
+
+# error codes that pandas test in CI
+# https://numpydoc.readthedocs.io/en/latest/validation.html#built-in-validation-checks
+NUMPYDOC_BASE_ERR_CODE = {
+    *("GL01", "GL02", "GL03", "GL04", "GL05", "GL06", "GL07", "GL09", "GL10"),
+    *("SS02", "SS04", "SS05", "PR03", "PR04", "PR05", "PR10", "EX04", "RT01"),
+    *("RT04", "RT05", "SA02", "SA03"),
+}
```

Comment:
Pay attention to the ERROR CODES that are being tested. The codes were taken from [the file](https://github.com/pandas-dev/pandas/blob/7d4757b4deb851bb44ab6bb20cdc404fa13fffcf/ci/code_checks.sh#L224), but I added two more to them (added for subjective sensations of readability) - GL01 and GL02.

An agreement should be reached. Cc @vnlitvinov @YarShev @devin-petersohn @prutskov .

Reply:
Do you have links describing `GL01` and `GL02` then?

Reply:
Yes, https://numpydoc.readthedocs.io/en/latest/validation.html#built-in-validation-checks

Reply:
Relevant piece:
```
    "GL01": "Docstring text (summary) should start in the line immediately "
            "after the opening quotes (not in the same line, or leaving a "
            "blank line in between)",
    "GL02": "Closing quotes should be placed in the line after the last text "
            "in the docstring (do not close the quotes in the same line as "
            "the text, or leave a blank line between the last text and the "
            "quotes)",
```

I'm fine with adding those checks, we follow them now in most places anyway.

Reply:
1) `GL04`, `EX04` are not presented in the guide. What is it?
2) Why are some of the numpydoc error codes missed?

Reply:
> 1. `GL04`, `EX04` are not presented in the guide. What is it?


Nice catch! These codes are defined by pandas itself and are absent in numpydoc.

> 2. Why are some of the numpydoc error codes missed?


Some codes are contrary to the standard (for example- `GL01`), some will lead to too much strictness in my opinion(for example - `PR08`).

Reply:
I suggest adding the following codes to current ones:
- GL08 if it doesn't require a docstring for a nested function.
- SS03
- PR01
- PR02

Reply:
I would prefer to add the following codes:

- PR07/RT03 - In my opinion, all parameters/returns should have a description.
- PR08/PR09 - This will ensure uniform style, I think. (It's already use for return values RT04/RT05)

Reply:
- On PR07/RT03, as we discussed on our previous weekly meeting, if a function is simple and its parameters and return values are intuitive, we can miss their description to not duplicate line of the type and line of the description.
- On PR08/PR09, if these checks do not contrast PR07/RT03 (there is no error that description is missed), we can add them.

Reply:
Added next codes: `GL08`, `SS03`, `PR01`, `PR02`, `PR08`, `PR09`.

Reply:
I am okay with this.